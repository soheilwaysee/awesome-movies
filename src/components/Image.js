import React from "react";
import config from "../config";

const Image = ({ alt, src, className }) => (
  <img
    {...(src
      ? { src: `${config.IMAGE_BASE_URL}${src}` }
      : { src: "/images/placeholder.png" })}
    data-testid="image"
    alt={alt}
    className={className}
  />
);

export default Image;
