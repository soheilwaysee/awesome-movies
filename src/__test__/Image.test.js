import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "@testing-library/react";
import "jest-dom/extend-expect";
import Image from "../components/Image";
import { Switch, Route, MemoryRouter } from "react-router-dom";

const App = () => (
  <MemoryRouter initialEntries={["/one"]} initialIndex={0}>
    <Switch>
      <Route
        path="/one"
        component={() => <div data-testid="one">one is matched</div>}
      />
      <Route
        path="/two"
        component={() => <div data-testid="two">two is matched</div>}
      />
    </Switch>
  </MemoryRouter>
);

afterEach(cleanup);

it("renders personalized greeting", async () => {
  // Render new instance in every test to prevent leaking state
  const { getByTestId } = render(<App />);
  expect(getByTestId("one")).toBeInTheDocument();

  //   const { getByTestId, unmount, container } = render(<Image />);
  //   const imageNode = getByTestId("image");
  //   expect(imageNode).toBeInTheDocument();
  //   unmount();
  //   expect(imageNode).not.toBeInTheDocument();
  //   expect(imageNode).toMatchSnapshot();
  //   cleanup();
  //   expect(document.body.innerHTML).toBe("");
});
// it("renders personalized greeting", async () => {
//   // Render new instance in every test to prevent leaking state
//   const { getByText } = render(<Image name="Satoshi" />);

//   await waitForElement(() => getByText(/hello Satoshi/i));
// });
