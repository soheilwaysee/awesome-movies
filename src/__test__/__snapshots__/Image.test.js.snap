// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders personalized greeting 1`] = `
<img
  data-testid="image"
  src="/images/placeholder.png"
  username="soheil"
/>
`;
